postgres/datadog_checks/postgres/autodiscovery-script.py

```diff
@@ -0,0 +1,193 @@
+# Proof of concept script for Postgres db autodiscovery.
+# Pass a host to discover databases on that host.
+
+from typing import List, Callable
+import psycopg2
+import sys
+import os
+import threading
+import time
+from six import iteritems
+from  datetime import datetime
+from datadog import initialize, statsd
+
+sys.path.append(os.path.abspath("/home/ec2-user/dd/integrations-core/datadog_checks_base"))
+
+from util import fmt, get_schema_field
+from datadog_checks.base.utils.discovery import Discovery
+from datadog_checks.base import AgentCheck
+
+# sys.path.append(os.path.abspath("/home/ec2-user/dd/integrations-core"))
+from relationsmanager import RelationsManager, RELATION_METRICS, REL_METRICS
+from connections import MultiDatabaseConnectionPoolLimited
+
+
+AUTODISCOVERY_QUERY: str = """select {columns} from pg_catalog.pg_database where datistemplate = false;"""
+
+options = {
+    'statsd_host':'127.0.0.1',
+    'statsd_port':8125
+}
+
+initialize(**options)
+
+class PostgresAutodiscovery(Discovery): 
+    def __init__(self, host: str, max_conn: int) -> None:
+        super(PostgresAutodiscovery, self).__init__(self._get_databases, include={'.*': 10}, exclude=[], interval=60)
+        self.host = host
+        relations_config = [{'relation_regex': '.*'}]
+        self._relations_manager = RelationsManager(relations_config)
+        self._conn_pool = MultiDatabaseConnectionPoolLimited(self._connect, max_conn)
+        self.default_ttl = 60000
+        # get once to cache dbs
+        self.get_items()
+        # self._conn_pool = psycopg2.ThreadedConnectionPool(minconn=0, maxconn=maxconn, user="postgres", password="p0stgres", port="5432")
+    
+    def get_items(self) -> List[str]:
+        """
+        Get_items() from parent class returns a generator with four objects:
+            > yield pattern, key(item), item, config
+        This function takes the item of interest (dbname) from this four-tuple
+        and returns the full list of database names from the generator.
+        """
+        items = list(super().get_items())
+        items_parsed = [item[1] for item in items]
+        return items_parsed
+   
+    def _connect(self, dbname: str = None) -> None:
+        # Use ident method
+        connection_string = "host="+self.host+" user=postgres password=p0stgres"
+        if dbname is not None:
+            connection_string += " dbname=" + dbname
+        conn = psycopg2.connect(connection_string)
+
+        # print("connected")
+        return conn
+
+    def _get_autodiscovery_query(self) -> str:
+        autodiscovery_query = AUTODISCOVERY_QUERY.format(columns=', '.join(['datname']))
+        return autodiscovery_query
+
+    def _get_databases(self) -> List[str]:
+        conn = self._conn_pool.get_connection('postgres', self.default_ttl)
```

Comment:
I realize this is POC code, but curious about how you'll handle when this returns None? (which is what it seems will happen when max_connections is exceeded)